routes/dao_coin_exchange.go

```diff
@@ -346,6 +346,60 @@ func buildDAOCoinLimitOrderResponse(
 // Helper functions to calculate price and exchange rates for DAO coin limit orders
 ///////////////////////////////////////////////////////////////////////////////////
 
+// GetBestAvailableExchangeRateCoinsToBuyPerCoinToSell computes the best available decimal string exchange rate at which
+// the market is able to exchange one base unit of the selling coin pair for the buying coin. Since we are interested
+// in computing the best exchange rate for the selling coin, the denominator for the output will always be the selling coin.
+//   Example: given buying coin B, and selling coin S, an output exchange rate of "1.5" implies an exchange rate of
+//            (1.5 coin B) per (1 coin S).
+// This function can support any arbitrary coin pair, but is most useful for markets where one coin is always considered
+// the denominating coin (ex: DAO coin <> DESO). In such cases, this computes the best available ask price.
+func GetBestAvailableExchangeRateCoinsToBuyPerCoinToSell(
+	fes *APIServer,
+	utxoView *lib.UtxoView,
+	buyingCoinPKID *lib.PKID,
+	sellingCoinPKID *lib.PKID,
+) (string, error) {
+	// This is relatively inefficient as it pulls all open orders from one side of the book. We need to call this function
+	// as an abstraction for core behavior because it performs useful filtering of soft-deleted orders, and merges orders
+	// from the mempool and db. Long term, it will be worth further optimizing it further to support pagination, so it can
+	// return the top 1 order.
+	orders, err := utxoView.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(buyingCoinPKID, sellingCoinPKID)
+	if err != nil {
+		return "", err
+	}
+	if len(orders) == 0 {
+		// It's OK if there are no orders on the book that allow us to exchange the coin pair. We default the exchange
+		// rate to 0 in this case
+		return "0", nil
+	}
+
+	bestOrderSellingOneBaseUnitOfTheSellingCoin := orders[0]
```

Comment:
nit: You could track the best price instead of the best order. That would save you from having to make copies of the order entry in the loop.